src/tools/optimization-options.h

```diff
@@ -315,18 +317,38 @@ struct OptimizationOptions : public ToolOptions {
         //   --foo --pass-arg=foo@ARG
         Options::Arguments::Optional,
         [this, p](Options*, const std::string& arg) {
+          PassInfo info(p);
           if (!arg.empty()) {
-            if (passOptions.arguments.count(p)) {
-              Fatal() << "Cannot pass multiple pass arguments to " << p;
-            }
-            passOptions.arguments[p] = arg;
+            info.argument = arg;
           }
-          passes.push_back(p);
+
+          passes.push_back(info);
         },
         PassRegistry::get()->isPassHidden(p));
     }
   }
 
+  void addPassArg(const std::string& key, const std::string& value) override {
```

Comment:
This could benefit from a comment as to what it is doing overall (looking for the last appearance etc.).

Reply:
I added a few comments to the function.